### PR TITLE
Handle zero values for the last recieved dates, better logging, fix me breaking the device info response

### PR DESCRIPTION
**Pull request by grahamgilbert**

### CODE REVIEW

File: director/background_tasks.go [hunk 1]
```diff
@@ -111,11 +111,10 @@ func pushAll() error {
 	var dbDevices []types.Device
 	now := time.Now()
 
-	DelaySeconds, HalfDelaySeconds := getDelay()
+	DelaySeconds, _ := getDelay()
```
Comment (discentem) [inline director/background_tasks.go:114]: Why does `getDelay()` return two values? Can we return just one value and do `(getDelay() / 2)` in place? 

- I realize it's possible a lot of calls to getDelay might need to be changed. :/ 
- If you are willing do the above, can we make the function name more description? What is the context for this delay?

File: director/background_tasks.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func pushAll() error {
 			}
 		}
 		// This contrived bit of logic is to handle devices that don't have a LastScheduledPush set yet
-		if !dbDevice.LastScheduledPush.Before(lastCheckinDelay) {
+		if !dbDevice.LastScheduledPush.IsZero() || dbDevice.LastCertificateList.IsZero() || dbDevice.LastProfileList.IsZero() || dbDevice.LastSecurityInfo.IsZero() || dbDevice.LastDeviceInfo.IsZero() {
```
Comment (discentem) [inline director/background_tasks.go:143]: LastScheduledPush is time.Time and IsZero is nil value for time?

Comment (grahamgilbert) [reply director/background_tasks.go:143]: Yes - i.e. has one ever happened?

File: director/background_tasks.go [hunk 3]
```diff
@@ -111,11 +110,15 @@ func pushAll() error {
 	var dbDevices []types.Device
 	now := time.Now()
 
-	DelaySeconds, HalfDelaySeconds := getDelay()
+<<<<<<< HEAD
+	DelaySeconds, _ := getDelay()
+=======
+	DelaySeconds := getDelay()
+	HalfDelaySeconds := DelaySeconds / 2
+>>>>>>> fix_profile_list_comparison
```
Comment (discentem) [inline director/background_tasks.go]: Merge conflict?

**Comment by discentem** [comment]
Fix failing build and you should be all set

**Pull request merged**